File: 101/exercise_rework/easy_4/number_to_string.rb
# number_to_string.rb

NUMBER_STRING = { 1 => '1', 2 => '2', 3 => '3', 4 => '4',
                  5 => '5', 6 => '6', 7 => '7', 8 => '8',
                  9 => '9', 0 => '0' }

def integer_to_string(int)
  int.digits.reverse.map { |digit| NUMBER_STRING[digit] }.join
end

File: 210/js_fundamentals_exercises/basics/number_to_string.js
// number_to_string.js

const DIGIT_TO_STRING_DICTIONARY = ['0', '1', '2', '3', '4', '5', '6', '7', '8', '9'];

function integerToString(num) {
  if (num === 0) { return num }
  var digitArray = [];

  while (num >= 1) {
    digit = num % 10;
    digitArray.push(digit);
    num = Math.floor(num / 10)
  }
  return convertDigits(digitArray).reverse().join('');
}

function convertDigits(arr) {
  return arr.map( digit => DIGIT_TO_STRING_DICTIONARY[digit] )
}

console.log(integerToString(4321));      // "4321"
console.log(integerToString(0));         // "0"
console.log(integerToString(5000));      // "5000"